src/unix/linux-core.c

```diff
@@ -1015,14 +1016,281 @@ uint64_t uv_get_total_memory(void) {
 }
 
 
-static uint64_t uv__read_cgroups_uint64(const char* cgroup, const char* param) {
-  char filename[256];
+#define PROC_SELF_MOUNTINFO "/proc/self/mountinfo"
+#define PROC_SELF_CGROUP "/proc/self/cgroup"
+#define CGROUPS_VERSION_UNKNOWN 0x0
+#define CGROUPS_VERSION_1 0x1
+#define CGROUPS_VERSION_2 0x2
+#define CGROUPS_V2_CTRL_PREFIX "0::"
+#define CGROUPS_V2_CTRL_PREFIX_LEN 3
+#define CGROUPS_V2_NO_LIMIT_VALUE "max"
+#define CGROUPS_BUF_SIZE 16384
+
+
+/*
+ * Holds information about a given cgroups subsystem.
+ */
+typedef struct {
+  /* cgroups version, one of CGROUPS_VERSION_*. */
+  uint8_t cgroups_version;
+  /* Path in which param files are found for a subsystem. */
+  char* path;
+} uv__cgroups_subsystem_info_t;
+
+
+/*
+ * Given a `separator`-delimited string `haystack`, return 1 if `needle` exactly
+ * matches at least one of the elements in that sequence.
+ */
+static int uv__find_in_delimited_string(const char* haystack, const char* needle, char separator) {
+  char* haystack_mutable;
+  char* candidate;
+  char* haystack_ptr;
+  if (needle == NULL || strlen(needle) > strlen(haystack))
+    return 0;
+  haystack_mutable = malloc(strlen(haystack) + 1);
+  strcpy(haystack_mutable, haystack);
+  haystack_ptr = haystack_mutable;
+  do {
+    candidate = strsep(&haystack_ptr, &separator);
```

Comment:
AFAICT `strsep` expects the delimiter to be a null terminated string rather than the address of a char.